dotnet/src/Connectors/Connectors.HuggingFace/Core/HuggingFaceClient.cs

```diff
@@ -134,7 +134,8 @@ internal HttpRequestMessage CreatePost(object requestData, Uri endpoint, string?
         var endpoint = this.GetTextGenerationEndpoint(modelId);
         var request = this.CreateTextRequest(prompt, executionSettings);
 
-        using var activity = ModelDiagnostics.StartCompletionActivity(endpoint, modelId, this.ModelProvider, prompt, executionSettings);
+        var huggingFaceExecutionSettings = HuggingFacePromptExecutionSettings.FromExecutionSettings(executionSettings);
```

Comment:
Should the LogTextGenerationUsage later in the method use this one instead of the original?

Reply:
Good catch! Yes, in theory. But it won't change anything ATM. I will change it for consistency.